src/uTensor/core/quantizationPrimitives.cpp

```diff
@@ -9,23 +9,23 @@ QuantizationParams::QuantizationParams(const int32_t* zp, const float* scale,
     : _zp(zp), _scale(scale), _num_channels(num_channels) {}
 
 QuantizationParams::~QuantizationParams() {}
-int32_t QuantizationParams::get_zeroP_for_channel(int i) { return 0; }
-float QuantizationParams::get_scale_for_channel(int i) { return 0; }
+int32_t const QuantizationParams::get_zeroP_for_channel(int i) const { return *_zp; }
```

Comment:
```suggestion
const int32_t QuantizationParams::get_zeroP_for_channel(int i) const { return 0; }
```

Reply:
Default quantization params should always return zero as nothing is set. Otherwise potential nullptr dereference

Reply:
Are you suggesting to use `PerTensorQuantizationParams` instead?
If `QuantizationParams` is an abstraction, then, it only represent a small sets of quantization scheme. Not all quantization scheme are can be represented in zero-points and scales.

Reply:
@mbartling actually, tensorBase.cpp:66&71 may have caused object slicing since the `_qnt_params` is declared as a base class.

Reply:
I know, this is meant to be a temporary hack since there are a variety of quantization schemes and sub 8 bit schemes. The default quantization is `assume zero` which will not affect any of the generic operators. The specialization here is either per tensor quantization or per channel. Eventually, I will flesh out a proper lightweight system for handling the other various cases, but for now the status quo is the symmetric form of either per tensor quantization or per channel